### PR TITLE
fast path for mongodb batches

### PR DESCRIPTION
**Pull request by BenLubar**
The upgrade for 1.0.0 looks like this: (I stopped it because I didn't want it to run for days)
image

With this patch, it looks like this:
image

(at the time of posting, the upgrade's last two lines are `Creating users:notvalidated done!` and `Creating Global moderators group`)

See also: akhoury/nodebb-plugin-import#176

### CODE REVIEW

**Comment by akhoury** [comment]
+1 agreed - @barisusakli maybe there is a similar solution for redis.

**Comment by barisusakli** [comment]
I think http://redis.io/commands/zscan is similar

**Comment by barisusakli** [comment]
This looks promising ill see if I can add `db.processSortedSet` to the redis code as well.

**Comment by akhoury** [comment]
awesome !

File: src/database/mongo/sorted.js [hunk 1]
```diff
@@ -533,4 +533,40 @@ module.exports = function(db, module) {
...
+			function() {
+				return !done;
+			},
+			function(next) {
+				cursor.next(function(err, item) {
+					if (err) {
+						return next(err);
+					}
+					if (item === null) {
+						done = true;
```
Comment (barisusakli) [inline src/database/mongo/sorted.js]: I think we are missing a `return next()` here I will add that in

Comment (BenLubar) [inline src/database/mongo/sorted.js]: If you `return next()`, it won't run the `process` function for the last batch.

Comment (barisusakli) [inline src/database/mongo/sorted.js]: Hmm what happens when a sorted set has 1 element and batchsize is 30.

I think it never calls process on that 1 element.

Comment (barisusakli) [inline src/database/mongo/sorted.js]: Nevermind there is the ids.length ===0 check

Comment (BenLubar) [inline src/database/mongo/sorted.js]: The first iteration gets the element and puts it into the ids array. The second iteration gets null for item, so it sets done and calls process, then calls next from the callback for process.

**Pull request merged**